server/apps/immich/src/api-v1/asset/asset.controller.ts

```diff
@@ -134,6 +140,20 @@ export class AssetController {
     return this.assetService.downloadFile(query, res);
   }
 
+  @Get('/download-library')
+  async downloadLibrary(
+    @GetAuthUser() authUser: AuthUserDto,
+    @Query(new ValidationPipe({ transform: true })) dto: DownloadDto,
+    @Response({ passthrough: true }) res: Res,
+  ): Promise<any> {
+    const { stream, fileName, fileSize, fileCount, complete } = await this.assetService.downloadLibrary(authUser, dto);
+    res.attachment(fileName);
```

Comment:
Do you have any good resources about the attachment property of the response parameter? I've never used it before so I just want to learn more about it

Reply:
Nothing too special about it, it just convenient method to set the content disposition header, which does all the magic.

http://expressjs.com/en/5x/api.html#res.attachment
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition